pkg/runner/job_status.go

```diff
@@ -99,20 +91,24 @@ func (jr *JobRunner) buildTestStepStatus(coordinates job.TestStepCoordinates) (*
 		return nil, fmt.Errorf("could not fetch events associated to test step %s: %v", coordinates.TestStepLabel, err)
 	}
 
-	// Build a map of events that should not be rendered as part of the status (e.g. routing events)
-	skipEvents := make(map[event.Name]struct{})
-	for _, eventName := range TargetRoutingEvents {
-		skipEvents[eventName] = struct{}{}
-	}
-	filteredTestEvents := []testevent.Event{}
+	var stepEvents, targetEvents []testevent.Event
 	for _, event := range testEvents {
-		if _, skip := skipEvents[event.Data.EventName]; !skip {
-			filteredTestEvents = append(filteredTestEvents, event)
+		if event.Data.Target == nil {
+			// we don't want target routing events in step events, but we want
+			// them in target events below
+			if _, skip := targetRoutingEvents[event.Data.EventName]; skip {
+				continue
+			}
```

Comment:
Can this ever happen, or are routing events with nil targets always a bug?

Reply:
it should never happen, I can add a log.Warning here

Reply:
@tfg13 do you want me to add that warning? Otherwise is this good to go?

Reply:
updated with warning, this is definitely something we want to detect